File: frontend/src/domain/job.ts
export type JobStatus = 'PENDIENTE' | 'PROCESANDO' | 'COMPLETADO' | 'FALLIDO'
export interface Job { job_id: string; text: string; status: JobStatus; sentiment: string | null; keywords: string[] | null; error: string | null; created_at: string; updated_at: string }
export interface CreatedJob { job_id: string; status: JobStatus }

File: python-service/app/domain/job.py
from dataclasses import dataclass
from datetime import datetime
from enum import StrEnum
from uuid import UUID


class JobStatus(StrEnum):
    PENDING = "PENDIENTE"
    PROCESSING = "PROCESANDO"
    COMPLETED = "COMPLETADO"
    FAILED = "FALLIDO"


@dataclass(frozen=True)
class Job:
    id: UUID
    text: str
    status: JobStatus
    sentiment: str | None
    keywords: list[str] | None
    error: str | None
    created_at: datetime
    updated_at: datetime

